flips/20220111-execution-effort.md

```diff
@@ -0,0 +1,555 @@
+| ValueExists                | function calls                           | Checks if a certain register exists                                                                         |
+| ValueDecoded               | function calls                           | Cadence callback after a value was decoded into a cadence type                                              |
+| ValueEncoded               | function calls                           | Cadence callback after a cadence type was encoded into a cadence json                                       |
+
+### Data analysis
+
+The large amount of different weights chosen was intentional. This allows for better insight into which weights should actually be used.
+
+The Following chapters describe how the linear model fitting was done due to a significant amount of outliers in the data and how certain weights were eliminated, because they were either redundant (correlated with other weights), insignificant (had small or no impact on the correlation), or had a lot of noise.
+
+A note on the `time to execute [ms]` vs `Execution effort` graphs. The closer the data is to the red dashed line which represents `time to execute = Execution effort` the better the fit of the model. Data that is to the right side of the graph, represents transactions that took longer than expected, and were thus charged less than they should be (but some variance and outliers are expected here). Data to the left side of the graph represents transactions where the execution did not take as long as the model predicted, and as a consequence those transactions will be charged more then they should be.
+
+#### Weight correlation
+
+Before using the data in a linear model, its important to note that the rank of the matrix of all weight data <!-- $M$ --> <img style="transform: translateY(0.1em); background: white;" src="./20220111-execution-effort/eq/SmblEwKco6.svg"> is smaller than the number of weights. This means that some wights are correlated to others. These are asy to spot by checking the correlation matrix where only the entries with 1 or -1 are highlighted.
+
+![Correlation1](./20220111-execution-effort/correlation1.png)
+
+The pairs that are correlated are:
+
+- `ProgramChecked` / `ProgramParsed`
+- `GetCode` / `GetAccountContractCode`
+- `ValueExists` / `CreateAccount`
+
+These pairs of weights were cross checked with the implementation to confirm that they are always called together.
+
+The columns with weights `ProgramChecked`, `GetCode` and `ValueExists` were removed from the weight data matrix <!-- $M$ --> <img style="transform: translateY(0.1em); background: white;" src="./20220111-execution-effort/eq/SmblEwKco6.svg">.
+
+#### Outliers and robust linear model fitting
+
+The following is a linear model fit on the data and plotting the data on a graph of execution time taken (in milliseconds) vs execution effort.
+
+![DemoFit](./20220111-execution-effort/demo-fit.png)
```

Comment:
A couple request: 
* could you please include a legend, to clearly identify the `red` as the desired output for a perfect model; and the blue as the predictions of our linear model. 
* Can you reflect the density in your graph. The data points in the around the model completely cover the domain. There could be significant differences in density, which are important to evaluate the performance of our model "by human inspection". 

Lets take a look at the model's performance for fast transactions (short 
To me, it looks there is a very distinct different